### PR TITLE
Add release notes for 1.71.0

### PR DESCRIPTION
**Pull request by cuviper**
r? @Mark-Simulacrum 
cc @rust-lang/release

### CODE REVIEW

File: RELEASES.md [hunk 1]
```diff
@@ -1,3 +1,141 @@
+Version 1.71.0 (2023-07-13)
+==========================
+
+<a id="1.71.0-Language"></a>
+
+Language
+--------
+
+- [Remove misleading target feature aliases, inappropriately connected to AVX-512.](https://github.com/rust-lang/rust/pull/107707/)
```
Comment (taiki-e) [inline RELEASES.md:9]:
```suggestion
```

avx512_target_feature is unstable.

File: RELEASES.md [hunk 2]
```diff
@@ -1,3 +1,141 @@
...
+- [Promote `loongarch64-unknown-linux-gnu` to Tier 2 with host tools.](https://github.com/rust-lang/rust/pull/110936/)
+
+Refer to Rust's [platform support page][platform-support-doc]
+for more information on Rust's tiered platform support.
+
+<a id="1.71.0-Libraries"></a>
+
+Libraries
+---------
+- [Remove structural match from `TypeId`.](https://github.com/rust-lang/rust/pull/103291/)
```
Comment (taiki-e) [inline RELEASES.md:43]: The structural-match traits are unstable, but IIUC this affects stable rust users (https://github.com/sval-rs/value-bag/issues/62), so I think it probably makes sense to move this to the compatibility note and mention that code that uses a constant of TypeId in a pattern will potentially be broken.

File: RELEASES.md (hunk 1 above)
Comment (cuviper) [reply RELEASES.md:9]: Ah, OK, I thought it would apply stability by the non-AVX names, but those are still unstable on the same feature gate.

**Pull request merged**